src/tokens/ERC1155.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity >=0.8.0;
+
+import {ERC1155TokenReceiver} from "./ERC1155TokenReceiver.sol";
+
+/// @notice Modern and gas efficient ERC1155 implementation.
+/// @author Modified from 0xsequence (https://github.com/0xsequence/erc-1155)
+abstract contract ERC1155 is ERC1155TokenReceiver {
```

Comment:
Should this also be an ERC1155TokenReceiver @transmissions11 ? Don't have default impls in the ERC1155TokenReceiver as well, should I add?

Reply:
wait why would it be a reciver? the contract shouldnt accept its on tokens right?

Reply:
in terms of default impls nah lets just keep an interface